zabbixctl/Zabbix.py

```diff
@@ -1,104 +1,103 @@
-from pyzabbix import ZabbixAPI, ZabbixAPIException
+"""Contains all the classes and functions associated with the Zabbix object
+type"""
+# Standard Lib
 import getpass
 import logging
 from utils import Cache
+from urlparse import urlparse, urljoin
+
+# Packages
+from pyzabbix import ZabbixAPI, ZabbixAPIException
 from requests.exceptions import HTTPError, ConnectionError
 
-c = Cache('/tmp/zabbix.cache')
+log = logging.getLogger(__name__)
+
+
+class ZabbixError(Exception):
+    pass
+
+
+class ZabbixNotAuthorized(ZabbixError):
+    pass
 
 
 class Zabbix(object):
-    """
 
-    """
-    status = True
-    error = None
+    API_PATH = 'zabbix'
 
     def __init__(self, host, noverify=False, cacert=None, http=False, timeout=30):
         """
-
-        :param host:
-        :param noverify:
-        :param cacert:
-        :param http:
-        :param timeout:
-        :return:
+        Initializes a Zabbix instance
+        :param host: hostname to connect to (ex. zabbix.yourdomain.net)
+        :param noverify: turns off verification
+        :param cacert: the certificate authority to use
+        :param http: flag to use http over https
+        :param timeout: API timeout parameter
+        :return: Zabbix instance
         """
-        self.logger = logging.getLogger('zabbixctl')
-        self.zabbix_url = host
-        protocol = 'http' if http else 'https'
 
-        self.zabbix_url = '{0}://{1}/zabbix'.format(protocol, host)
+        self.cache = Cache('/tmp/zabbix.cache')
+        self.host = host
 
-        self.logger.debug(self.zabbix_url)
+        zabbix_url = urlparse(
+            scheme='http' if http else 'https',
+            url=urljoin(host, self.API_PATH)
+        ).geturl()
```

Comment:
urlparse() parses a string into parts.  I think you were looking for urlunparse[1]:
```
zabbix_url = urlunparse([
    'http' if http else 'https',
    host,
    self.API_PATH,
    '', '', ''
    ]
)
```
1. https://docs.python.org/2/library/urlparse.html#urlparse.urlunparse